src/librustc_lint/array_into_iter.rs

```diff
@@ -48,14 +48,31 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for ArrayIntoIter {
             // argument.
             let receiver_arg = &args[0];
 
-            // Test if the original `self` type is an array type.
-            match cx.tables.expr_ty(receiver_arg).kind {
-                ty::Array(..) => {}
-                _ => return,
+
+            fn is_array(ty: &ty::TyS<'_>) -> bool {
+                if let ty::Array(..) = ty.kind { true } else { false }
```

Comment:
```suggestion
                matches!(ty.kind, ty::Array(..))
```

Don't forget the feature gate.

Reply:
What a great macro! I missed that it was already merged. Thanks for the hint. The feature gate was even already there for `librustc_lint` :)